File: service/src/main/java/net/huray/example/template/config/BeanConfig.java
package net.huray.example.template.config;

import com.querydsl.jpa.impl.JPAQueryFactory;
import org.modelmapper.ModelMapper;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;

@Configuration
public class BeanConfig {

    @PersistenceContext
    EntityManager em;

    @Bean
    public JPAQueryFactory jpaQueryFactory() {
        return new JPAQueryFactory(em);
    }

    @Bean
    public ModelMapper modelMapper() {
        return new ModelMapper();
    }
}

File: service/src/main/kotlin/net/huray/backend/minuting/config/BeanConfig.kt
package net.huray.backend.minuting.config

import com.querydsl.jpa.impl.JPAQueryFactory
import org.modelmapper.ModelMapper
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import javax.persistence.EntityManager
import javax.persistence.PersistenceContext

@Configuration
class BeanConfig(
    @PersistenceContext var em: EntityManager
) {

    @Bean
    fun jpaQueryFactory(): JPAQueryFactory = JPAQueryFactory(em)

    @Bean
    fun modelMapper(): ModelMapper = ModelMapper()

}